### DISCUSSION TITLE
Using GitHub is an abysmal experience.

### DISCUSSION THREAD

**Discussion by Tom-707** [community/community#31617] [Other Feature Feedback, Questions, & Ideas]
If your UX is bad, it doesn't matter how good your program is.

And yours is abysmal!!

Try searching for code. Totally un-usable!
Download a single file. Can't be done in a simple way!
A list of files in a repo shows you what needs to be done with each file. Not a meaningful description of what the file is used for.
The list goes on and on...

The developers need to do some serious design changes. Otherwise somebody else is going to come along with a much better application, and GitHub will be instantly discarded like the garbage it is.

**Comment by Quanloh** [comment]
I am not an experienced user but here's my 2 cents.

1. I totally agree that everything on earth needs improvement.
2. Somebody else will make a much better application/platform/UX, but Github will not be instantly discarded. 
It might slowly lose users, but all the learning material existed online will always point beginners to Github. 
And by the time user loss starts to become significant, Github will already finish "improving" their UX. 
3. I totally agree that everything on earth needs improvement.

**Comment by Kastakin** [comment]
While I might agree on the issues you are presenting this is not really a costructive way of expressing them.

Also you can downlaod single files easily: go to the file you want to download, right click on "Raw" and "Save Link as...", pick the right extension and you are done! :)

**Reply by Tom-707** [reply]
This is the MOST constructive way of expressing issues. Only THEN do people get the hint and take action.

"Also you can downlaod single files easily: go to the file you want to download, right click on "Raw" and "Save Link as...", pick the right extension and you are done!"
That's my point exactly!! How many steps does that take? How much easier would it be if you could just right click on the file and 'Save As...'? How hard is it to program that??!!

**Comment by aparatext** [comment]
I mean, how is GitHub supposed to know what is each file? A config file where you're forced to describe every single file? That would be bad UX. Also, it shows the last commit on each file, not what needs to be done. Searching for code isn't that bad. I can quickly search the entirety of GitHub for a code snippet and filter the results. Though the index is not always up to date on the project level. However, I do think some improvements can be made.

**Reply by Tom-707** [reply]
Each file should have a short description of what its function is. This should be put at the top in a comment, by the developer of that project.
GitHub could then read this and display it instead of (or in addition) to the last commit.
How much more meaningful would that be?!

Filtering a search result based on a specific repo, didn't work properly. Which is why, instead of punching a hole in the wall, I decided to start this discussion.

**Comment by mxrch** [comment]
Nobody needs to download only a single file, or if you need to, you get the raw link and wget it everywhere you want.
Either you git clone the repo, either you copy paste the lines you need in a file, but the more you will use Github, the more you'll find that Github UI/UX is actually very good and developer-friendly.

Concerning the search feature, I never had a single problem with several years of using it.
Also, they launched Github Code Search in private beta (https://cs.github.com), and you can search code with regex etc, it's just impressive.

**Reply by Tom-707** [reply]
That's my point. You can only download an entire repo with one click. How hard would it be to implement right click on a file and 'Save As...'?!
"Also, they launched Github Code Search in private beta (https://cs.github.com/), and you can search code with regex etc, it's just impressive."
Exactly! If their original search engine had worked well, they wouldn't need to implement a whole new service, just for searching.

**Reply by mxrch** [reply]
It's already working well. It's what we call : "new features".
Please don't be rude because you prefer to hate instead of learning how to use it.

**Reply by mxrch** [reply]
> You can only download an entire repo with one click. How hard would it be to implement right click on a file and 'Save As...'?!

Because Github is about git; managing git repositories, contribute, not about being a file hosting site.
